tests/test_elrond.py

```diff
@@ -0,0 +1,265 @@
+    UNKNOWN_INSTRUCTION    = 0x6D00
+    WRONG_CLA              = 0x6E00
+    SIGNATURE_FAILED       = 0x6E10
+    SIGN_TX_DEPRECATED     = 0x6E11
+    INVALID_ARGUMENTS      = 0x6E01
+    INVALID_MESSAGE        = 0x6E02
+    INVALID_P1             = 0x6E03
+    MESSAGE_TOO_LONG       = 0x6E04
+    RECEIVER_TOO_LONG      = 0x6E05
+    AMOUNT_TOO_LONG        = 0x6E06
+    CONTRACT_DATA_DISABLED = 0x6E07
+    MESSAGE_INCOMPLETE     = 0x6E08
+    WRONG_TX_VERSION       = 0x6E09
+    NONCE_TOO_LONG         = 0x6E0A
+    INVALID_AMOUNT         = 0x6E0B
+    INVALID_FEE            = 0x6E0C
+    PRETTY_FAILED          = 0x6E0D
+
+MAX_SIZE = 251
+ROOT_SCREENSHOT_PATH = Path(__file__).parent.resolve()
+
+
+@contextmanager
+def send_async_sign_message(client, ins, payload: bytes) -> Generator[None, None, None]:
+    payload_splited = [payload[x:x + MAX_SIZE] for x in range(0, len(payload), MAX_SIZE)]
+    p1 = P1.FIRST
+    if len(payload_splited) > 1:
+        for p in payload_splited[:-1]:
+            client.exchange(CLA, ins, p1, 0, p)
+            p1 = P1.MORE
+
+    with client.exchange_async(CLA,
+                                     ins,
+                                     p1,
+                                     0,
+                                     payload_splited[-1]):
+        yield
+
+
+class TestGetAppVersion:
+
+    def test_get_app_version(self, client, backend):
+        data = client.exchange(CLA, Ins.GET_APP_VERSION, P1.FIRST, 0, b"").data
+        version = data.decode("ascii").split('.')
+        assert version[0] == str(LEDGER_MAJOR_VERSION)
+        assert version[1] == str(LEDGER_MINOR_VERSION)
+        assert version[2] == str(LEDGER_PATCH_VERSION)
+
+
+class TestGetAppConfiguration:
+
+    def test_get_app_configuration(self, client, backend):
+        data = client.exchange(CLA, Ins.GET_APP_CONFIGURATION, P1.FIRST, 0, b"").data
+        assert len(data) == 14
+        assert data[0] == 0 or data[0] == 1                         # N_storage.setting_contract_data
+        # print(data[1])                                            # not to be taken into account anymore
+        # print(data[2])                                            # not to be taken into account anymore
+        assert data[3] == LEDGER_MAJOR_VERSION                      # LEDGER_MAJOR_VERSION
+        assert data[4] == LEDGER_MINOR_VERSION                      # LEDGER_MINOR_VERSION
+        assert data[5] == LEDGER_PATCH_VERSION                      # LEDGER_PATCH_VERSION
+        # assert int.from_bytes(data[6:10], byteorder="big") == 0     # bip32_account
+        # assert int.from_bytes(data[10:14], byteorder="big") == 0    # bip32_address_index
+
+
+class TestGetAddr:
+
+    def test_get_addr(self, client, backend):
+        account = 1
+        index = 1
+        payload = account.to_bytes(4, "big") + index.to_bytes(4, "big")
+        data = client.exchange(CLA, Ins.GET_ADDR, P1.NON_CONFIRM, P2.DISPLAY_HEX, payload).data
```

Comment:
no test with P1.CONFIRM?

Reply:
I'm only porting the tests in this PR, I'll maybe make changes after